entitlement/src/main/java/org/killbill/billing/entitlement/api/svcs/DefaultEntitlementApiBase.java

```diff
@@ -166,32 +166,20 @@ public Entitlement getEntitlementForId(final UUID entitlementId, final boolean i
     }
 
     public void pause(final UUID bundleId, @Nullable final LocalDate localEffectiveDate, final Iterable<PluginProperty> properties, final InternalCallContext internalCallContext) throws EntitlementApiException {
-
-        final DateTime effectiveDateTime = dateHelper.fromLocalDateAndReferenceTime(localEffectiveDate, internalCallContext.getCreatedDate(), internalCallContext);
-        final BaseEntitlementWithAddOnsSpecifier baseEntitlementWithAddOnsSpecifier = new DefaultBaseEntitlementWithAddOnsSpecifier(
-                bundleId,
-                null,
-                null,
-                effectiveDateTime,
-                effectiveDateTime,
-                false);
-        final List<BaseEntitlementWithAddOnsSpecifier> baseEntitlementWithAddOnsSpecifierList = new ArrayList<BaseEntitlementWithAddOnsSpecifier>();
-        baseEntitlementWithAddOnsSpecifierList.add(baseEntitlementWithAddOnsSpecifier);
```

Comment:
I am wondering if this shouldn't be passed into the context instead (see `resume`)?

Reply:
@pierre Yeah, I'm wondering that, too. But I thought it was the difference between `pause()` and `resume()`: That `DefaultEntitlementContext`for `pause()` should set `baseEntitlementWithAddOnsSpecifiers` to `null` ?

Reply:
`pause` and `resume` are very symmetrical (it's just an event being inserted in the subscription events stream, with block billing set/unset). Also, since this goes into `pluginContext`, it simply informs the entitlement plugins about which subscription is affected. So, I think I would pass it.

Reply:
@pierre fixed in this [pull request](https://github.com/killbill/killbill/pull/1861) by [this commit](https://github.com/killbill/killbill/pull/1861/commits/8488a2fd46e60474d35d5f8749be9c3bd05d8baa).